### PR TITLE
[AutoParallel] unify llama model && fix vpp unittest hang problem

### PR DESCRIPTION
**Pull request by deepllz**
### PR types

New features
### PR changes

Others
### Description

动静统一的llama组网，移除组网中诸多in_dynamic_mode的判断
Pcard-73145

### CODE REVIEW

File: test/auto_parallel/hybrid_strategy/semi_auto_parallel_llama_model.py [hunk 1]
```diff
@@ -684,34 +661,25 @@ def forward(
...
-                        position_ids,
-                        get_mesh(ipp),
-                        [dist.Shard(0), dist.Replicate()],
-                    )
-                    attention_mask = dist.reshard(
-                        attention_mask,
-                        get_mesh(ipp),
-                        [dist.Shard(0), dist.Replicate()],
-                    )
+            if ipp is None:  # without pp
```
Comment (zhiqiu) [inline test/auto_parallel/hybrid_strategy/semi_auto_parallel_llama_model.py]: the idx that needs reshard "pp_stage*[1,2,3, num_layers/pp_degree -1 ]"

Comment (deepllz) [reply test/auto_parallel/hybrid_strategy/semi_auto_parallel_llama_model.py]: Done

**Review by zhiqiu — APPROVED** [review]
LGTM

**Review by XieYunshen — APPROVED** [review]
LGTM for `  set_tests_properties(test_semi_auto_parallel_llama_model_vpp
                       PROPERTIES TIMEOUT "180" LABELS "RUN_TYPE=HYBRID")`
